QuietPlace/microphoneListener.js

```diff
@@ -35,20 +35,37 @@ export default class MicrophoneListener extends Component {
 	};
 
     componentDidMount(){
-        requestMicrophonePermission()
         const granted = PermissionsAndroid.check( PermissionsAndroid.PERMISSIONS.RECORD_AUDIO );
         if (granted) {
           console.log( "You can use the microphone" )
         }
         else {
           console.log( "Microphone permission denied" )
+          requestMicrophonePermission()
         }
         RNSoundLevel.start()
+        var count = 0;
+        var fiveSoundFrames = new Array(5)
+        var notificationPause = 20;
         RNSoundLevel.onNewFrame = (data) => {
             console.log('Sound level info', data)
             // If sound level is greater than slider value
 			//Data is measured from -160 to 0, but only using -100 to 0 for slider values
-            if(data.value >= this.state.value){
+            if (count == 5){
+              fiveSoundFrames.shift();
+              count--;
+            }
+            fiveSoundFrames.push(data.value);
+            count++;
+            const avg = fiveSoundFrames.reduce((p, c, _, a) => p + c/a.length, 0);
+            console.log(avg);
+            console.log(notificationPause);
+            if(notificationPause < 20){
+              notificationPause++;
+            }
+            if(count == 5 && avg >= this.state.value && notificationPause == 20){
+              fiveSoundFrames = new Array(5);
```

Comment:
I think the number of ticks in the average sound level detection should be decreased from 5 to 3.
3 feels more appropriate for spikes in volume without being as overbearing as 1.
5 feels a bit too forgiving.